bundle/jsky.app.ot/src/main/scala/jsky/app/ot/gemini/editor/targetComponent/details2/EphemerisEditor.scala

```diff
@@ -1,47 +1,164 @@
 package jsky.app.ot.gemini.editor.targetComponent.details2
 
-import java.awt.Color
+import java.awt.{Insets, GridBagConstraints, GridBagLayout, Color}
 import java.text.SimpleDateFormat
-import java.util.{ Date, TimeZone }
-import javax.swing.JLabel
+import java.util.{ TimerTask, Timer, Date, TimeZone }
+import javax.swing.{SwingConstants, BorderFactory, JPanel, JLabel}
 
 import edu.gemini.pot.sp.ISPNode
 import edu.gemini.shared.util.immutable.{ Option => GOption }
-import edu.gemini.spModel.core.{Declination, Angle, RightAscension, SiderealTarget, Coordinates, Target}
+import edu.gemini.spModel.core.{Ephemeris, NonSiderealTarget, Target}
 import edu.gemini.spModel.obs.context.ObsContext
 import edu.gemini.spModel.target.SPTarget
 import jsky.app.ot.gemini.editor.targetComponent.TelescopePosEditor
+import edu.gemini.shared.util.immutable.ScalaConverters._
 
 import scalaz._, Scalaz._
 
 class EphemerisEditor extends TelescopePosEditor with ReentrancyHack {
 
-  val df = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss z") <| { df =>
-    df.setTimeZone(TimeZone.getTimeZone("UTC"))
-  }
+  private[this] var target: Option[NonSiderealTarget] = None
 
-  val start, end, size = new JLabel <| { l =>
+  // TODO: stop the timer when the window closes, or something?
```

Comment:
I think there shouldn't be an issue with a partial write like you would see with updating a Double or Long, but my understanding was that to guarantee that changes made by one thread are seen by another you have to mark the reference `@volatile` or else synchronize it.